manifests/init.pp

```diff
@@ -328,7 +334,10 @@
   Stdlib::Absolutepath $puppetdir = $::foreman_proxy::params::puppetdir,
   String $puppetca_cmd = $::foreman_proxy::params::puppetca_cmd,
   String $puppet_group = $::foreman_proxy::params::puppet_group,
+  Enum['puppetca_hostname_whitelisting', 'puppetca_token_whitelisting'] $puppetca_provider = $::foreman_proxy::params::puppetca_provider,
```

Comment:
For other providers we've decided against an enum because we never know which plugins could be added. While unlikely, I would like users to be able to add their own providers. It would also be consistent with other `$*_provider` parameters.